### ISSUE TITLE
transformFromAst decorator in a wrong place

### ISSUE THREAD

**Issue by sotream** [babel/babel#5335]
### Input Code
[Demo repo](https://github.com/sotream/babylon_test)

```js
// Core
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { actions } from '../../actions/navigation';

const mapStateToProps = ({ navigation }) => ({ // eslint-disable-line arrow-body-style
	menuStatus: navigation.get('menuStatus')
});

const mapDispatchToProps = (dispatch) => ({ // eslint-disable-line arrow-body-style
	actions: bindActionCreators({ ...actions }, dispatch)
});

@connect(mapStateToProps, mapDispatchToProps)
export default class Home extends Component {
	render () {
		return (
			<section>
				<h1>Home container!</h1>
			</section>
		);
	}
}
```

### Babel Configuration (.babelrc, package.json, cli command)
All config files are in demo repository.

### Expected Behavior
Decorator should be top of default export

### Current Behavior
Decorator placed after after default export

### Possible Solution
?

### Context
I'm trying to parse js file with babylon. Make some modifications with babel-traverse (for example add a new import declaration) and generate js file back.

### Your Environment

| software         | version
| ---------------- | -------
| Babel            |  6.23.1
| node             | 7.5.0
| npm              | 4.1.2
| Operating System | Ubuntu 16.04

**Comment by loganfsmyth** [comment]
Thanks for the report! Looks like a duplicate of https://github.com/babel/babel/issues/4585 Looks like we had a proposed fix, but it was blocked waiting on feedback about what the expected behavior should be.

**Comment by sotream** [comment]
@loganfsmyth Thx for reply. Expected behavior - decorator should be top above default exports in my case.
